### ISSUE TITLE
IllegalStateException on Folia

### ISSUE THREAD

**Issue by asygamesperson** [Azuriom/AzLink#49]
Can someone please fix this error for azlink https://pastebin.com/FWj5YEKa
Thank you

**Comment by asygamesperson** [comment]
```
[10:15:08 ERROR]: Thread Region Scheduler Thread Azuriom/Plugin-Shop#1 failed main thread check: Accessing entity state off owning region's thread
java.lang.Throwable: null
        at io.papermc.paper.util.TickThread.ensureTickThread(TickThread.java:78) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftSkeleton.getHandle(CraftSkeleton.java:43) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftSkeleton.getHandle(CraftSkeleton.java:7) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftEntity.isInWorld(CraftEntity.java:940) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftEntity.isValid(CraftEntity.java:394) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.CraftRegionAccessor.lambda$getEntities$1(CraftRegionAccessor.java:315) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at org.bukkit.craftbukkit.CraftRegionAccessor.getEntities(CraftRegionAccessor.java:311) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at AzLink.jar/com.azuriom.azlink.bukkit.AzLinkBukkitPlugin.lambda$getWorldData$1(AzLinkBukkitPlugin.java:161) ~[AzLink.jar:?]
        at java.base/java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:214) ~[?:?]
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:515) ~[?:?]
        at java.base/java.util.stream.IntPipeline.sum(IntPipeline.java:473) ~[?:?]
        at AzLink.jar/com.azuriom.azlink.bukkit.AzLinkBukkitPlugin.getWorldData(AzLinkBukkitPlugin.java:162) ~[AzLink.jar:?]
        at AzLink.jar/com.azuriom.azlink.common.AzLinkPlugin.getServerData(AzLinkPlugin.java:159) ~[AzLink.jar:?]
        at AzLink.jar/com.azuriom.azlink.common.tasks.FetcherTask.lambda$fetch$1(FetcherTask.java:55) ~[AzLink.jar:?]
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.lambda$execute$0(FoliaGlobalRegionScheduler.java:47) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler$GlobalScheduledTask.run(FoliaGlobalRegionScheduler.java:179) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.tick(FoliaGlobalRegionScheduler.java:37) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.RegionizedServer.globalTick(RegionizedServer.java:293) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at io.papermc.paper.threadedregions.RegionizedServer$GlobalTickTickHandle.tickRegion(RegionizedServer.java:148) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:404) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:525) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[10:15:08 ERROR]: [AzLink] Unable to send data to the website: java.lang.IllegalStateException: Accessing entity state off owning region's thread
```

**Comment by asygamesperson** [comment]
```
[17:30:23 ERROR]: Thread Region Scheduler Thread #1 failed main thread check: Accessing entity state off owning region's thread
java.lang.Throwable: null
        at io.papermc.paper.util.TickThread.ensureTickThread(TickThread.java:78) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftPainting.getHandle(CraftPainting.java:62) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftPainting.getHandle(CraftPainting.java:11) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftEntity.isInWorld(CraftEntity.java:940) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftEntity.isValid(CraftEntity.java:394) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.CraftRegionAccessor.lambda$getEntities$1(CraftRegionAccessor.java:315) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at org.bukkit.craftbukkit.CraftRegionAccessor.getEntities(CraftRegionAccessor.java:311) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at AzLink.jar/com.azuriom.azlink.bukkit.AzLinkBukkitPlugin.lambda$getWorldData$1(AzLinkBukkitPlugin.java:161) ~[AzLink.jar:?]
        at java.base/java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:214) ~[?:?]
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:515) ~[?:?]
        at java.base/java.util.stream.IntPipeline.sum(IntPipeline.java:473) ~[?:?]
        at AzLink.jar/com.azuriom.azlink.bukkit.AzLinkBukkitPlugin.getWorldData(AzLinkBukkitPlugin.java:162) ~[AzLink.jar:?]
        at AzLink.jar/com.azuriom.azlink.common.AzLinkPlugin.getServerData(AzLinkPlugin.java:159) ~[AzLink.jar:?]
        at AzLink.jar/com.azuriom.azlink.common.tasks.FetcherTask.lambda$fetch$1(FetcherTask.java:55) ~[AzLink.jar:?]
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.lambda$execute$0(FoliaGlobalRegionScheduler.java:47) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler$GlobalScheduledTask.run(FoliaGlobalRegionScheduler.java:179) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.tick(FoliaGlobalRegionScheduler.java:37) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.RegionizedServer.globalTick(RegionizedServer.java:293) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at io.papermc.paper.threadedregions.RegionizedServer$GlobalTickTickHandle.tickRegion(RegionizedServer.java:148) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:404) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:525) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[17:30:23 ERROR]: [AzLink] Unable to send data to the website: java.lang.IllegalStateException: Accessing entity state off owning region's thread
```

**Comment by asygamesperson** [comment]
**I still get this error in console**

```
[00:15:21 ERROR]: Thread Region Scheduler Thread #0 failed main thread check: Accessing entity state off owning region's thread
java.lang.Throwable: null
        at io.papermc.paper.util.TickThread.ensureTickThread(TickThread.java:78) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftArmorStand.getHandle(CraftArmorStand.java:32) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftArmorStand.getHandle(CraftArmorStand.java:12) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftEntity.isInWorld(CraftEntity.java:940) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.entity.CraftEntity.isValid(CraftEntity.java:394) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at org.bukkit.craftbukkit.CraftRegionAccessor.lambda$getEntities$1(CraftRegionAccessor.java:315) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at org.bukkit.craftbukkit.CraftRegionAccessor.getEntities(CraftRegionAccessor.java:311) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at AzLink.jar/com.azuriom.azlink.bukkit.AzLinkBukkitPlugin.lambda$getWorldData$1(AzLinkBukkitPlugin.java:161) ~[AzLink.jar:?]
        at java.base/java.util.stream.ReferencePipeline$4$1.accept(ReferencePipeline.java:214) ~[?:?]
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:515) ~[?:?]
        at java.base/java.util.stream.IntPipeline.sum(IntPipeline.java:473) ~[?:?]
        at AzLink.jar/com.azuriom.azlink.bukkit.AzLinkBukkitPlugin.getWorldData(AzLinkBukkitPlugin.java:162) ~[AzLink.jar:?]
        at AzLink.jar/com.azuriom.azlink.common.AzLinkPlugin.getServerData(AzLinkPlugin.java:159) ~[AzLink.jar:?]
        at AzLink.jar/com.azuriom.azlink.common.tasks.FetcherTask.lambda$fetch$1(FetcherTask.java:55) ~[AzLink.jar:?]
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.lambda$execute$0(FoliaGlobalRegionScheduler.java:47) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler$GlobalScheduledTask.run(FoliaGlobalRegionScheduler.java:179) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.scheduler.FoliaGlobalRegionScheduler.tick(FoliaGlobalRegionScheduler.java:37) ~[folia-1.20.6.jar:?]
        at io.papermc.paper.threadedregions.RegionizedServer.globalTick(RegionizedServer.java:293) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at io.papermc.paper.threadedregions.RegionizedServer$GlobalTickTickHandle.tickRegion(RegionizedServer.java:148) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at io.papermc.paper.threadedregions.TickRegionScheduler$RegionScheduleHandle.runTick(TickRegionScheduler.java:404) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at ca.spottedleaf.concurrentutil.scheduler.SchedulerThreadPool$TickThreadRunner.run(SchedulerThreadPool.java:525) ~[folia-1.20.6.jar:1.20.6-5-d797082]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
[00:15:21 ERROR]: [AzLink] Unable to send data to the website: java.lang.IllegalStateException: Accessing entity state off owning region's thread
```

**Comment by MrMicky-FR** [comment]
It will be fixed soon, but please avoid bumping with the same error as it doesn’t help

**Comment by MrMicky-FR** [comment]
Fixed in AzLink v1.3.7
